visualization/src/interactor_style.cpp

```diff
@@ -1188,6 +1168,130 @@ pcl::visualization::PCLVisualizerInteractorStyle::OnKeyDown ()
+// Update the look up table displayed when 'u' is pressed
+void
+pcl::visualization::PCLVisualizerInteractorStyle::updateLookUpTableDisplay (bool add_lut)
+{
+  CloudActorMap::iterator am_it;
+  ShapeActorMap::iterator sm_it;
+  bool actor_found = false;
+
+  if (lut_actor_id_ != "")  // Search if provided actor id is in CloudActorMap or ShapeActorMap
+  {
+    am_it = cloud_actors_->find (lut_actor_id_);
+    if (am_it == cloud_actors_->end ())
+    {
+      sm_it = shape_actors_->find (lut_actor_id_);
+      if (sm_it == shape_actors_->end ())
+      {
+        PCL_WARN ("[updateLookUpTableDisplay] Could not find any actor with id <%s>!\n", lut_actor_id_.c_str ());
+        if (lut_enabled_)
+        {  // Remove LUT and exit
+          CurrentRenderer->RemoveActor (lut_actor_);
+          lut_enabled_ = false;
+        }
+        return;
+      }
+
+      // ShapeActor found
+      vtkSmartPointer<vtkProp> *act = & (*sm_it).second;
+      vtkSmartPointer<vtkActor> actor = vtkActor::SafeDownCast (*act);
+      if (!actor || !actor->GetMapper ()->GetInput ()->GetPointData ()->GetScalars ())
+      {
+        PCL_WARN ("[updateLookUpTableDisplay] id <%s> does not hold any color information!\n", lut_actor_id_.c_str ());
+        if (lut_enabled_)
+        {  // Remove LUT and exit
+          CurrentRenderer->RemoveActor (lut_actor_);
+          lut_enabled_ = false;
+        }
+        return;
+      }
+
+      lut_actor_->SetLookupTable (actor->GetMapper ()->GetLookupTable ());
+      lut_actor_->Modified ();
+      actor_found = true;
+    }
+    else
+    {
+      // CloudActor
+      CloudActor *act = & (*am_it).second;
+      if (!act->actor->GetMapper ()->GetLookupTable () && !act->actor->GetMapper ()->GetInput ()->GetPointData ()->GetScalars ())
+      {
+        PCL_WARN ("[updateLookUpTableDisplay] id <%s> does not hold any color information!\n", lut_actor_id_.c_str ());
+        if (lut_enabled_)
+        {  // Remove LUT and exit
+          CurrentRenderer->RemoveActor (lut_actor_);
+          lut_enabled_ = false;
+        }
+        return;
+      }
+
+      vtkScalarsToColors* lut = act->actor->GetMapper ()->GetLookupTable ();
+      lut_actor_->SetLookupTable (lut);
+      lut_actor_->Modified ();
+      actor_found = true;
+    }
+  }
+  else  // lut_actor_id_ == "", the user did not specify which cloud/shape LUT should be displayed
+  // Circling through all clouds/shapes and displaying last LUT found
+  {
+    for (am_it = cloud_actors_->begin (); am_it != cloud_actors_->end (); ++am_it)
+    {
+      CloudActor *act = & (*am_it).second;
+      if (!act->actor->GetMapper ()->GetLookupTable ())
+        continue;
+
+      if (!act->actor->GetMapper ()->GetInput ()->GetPointData ()->GetScalars ())
+        continue;
+
+      vtkScalarsToColors* lut = act->actor->GetMapper ()->GetLookupTable ();
+      lut_actor_->SetLookupTable (lut);
+      lut_actor_->Modified ();
+      actor_found = true;
```

Comment:
You should `break` here, otherwise search is continued even when a suitable actor is found.